sklearn/svm/src/liblinear/liblinear_helper.c

```diff
@@ -84,7 +85,7 @@ static struct feature_node **csr_to_sparse(double *values,
     for (i=0; i<shape_indptr[0]-1; ++i) {
         n = indptr[i+1] - indptr[i]; /* count elements in row i */
 
-        sparse[i] = malloc ((n+2) * sizeof(struct feature_node));
+        sparse[i] = malloc ((n+have_bias+1) * sizeof(struct feature_node));
```

Comment:
Do you know why there is a `+2` and not just `+1` (for bias) in the first place?

Reply:
It's because there has to be space for the sentinel too.